### ISSUE TITLE
Create Overview Infrastructure Mappings List View

### ISSUE THREAD

**Issue by priley86** [priley86/miq_v2v_ui_plugin#152]
Per the new UX, we will be able to visualize Infrastructure Mappings as a list view on the Overview Screen.

* include new Empty State view for when no mappings exist

**Comment by vconzola** [comment]
Mockups with the behavioral details for infrastructure mapping list view: https://docs.google.com/presentation/d/1iTLqYYMbbALD1BHcSPw9JVHvKfqK0J8X8HGcuigJ26U/edit?usp=sharing

**Comment by priley86** [comment]
Going to grab this one and knock it out now.

**Comment by priley86** [comment]
handing this over to @AllenBW ... we will need to gather the detail's for the Infra Mapping List View using a subsequent query (to get the source/target cluster names) **as the user expands the list item for the mapping**. That API is here:

```
POST localhost:3000/api/clusters?attributes=name,v_parent_datacenter
{
  "action": "query",
  "resources": [
    {"id": 5},
    {"id": 6}
  ]
}
returns
{
    "results": [
        {
            "href": "http://localhost:3000/api/clusters/5",
            "name": "rhevh1-Local",
            "id": "5",
            "v_parent_datacenter": ""
        },
        {
            "href": "http://localhost:3000/api/clusters/6",
            "name": "overcloud-Compute-gecols5lpz4o",
            "id": "6",
            "v_parent_datacenter": ""
        }
    ]
}
```

We could also potentially show the datacenter in the expansion details, but leave that up to @serenamarie125 @vconzola .

**Comment by priley86** [comment]
let me know if you need anything else @AllenBW ...

**Comment by AllenBW** [comment]
> The upshot of this is that wherever we display a cluster name in the UI we need to append its associated datacenter where it lives. I looked at a few different options and prefer that we show it as "datacenter_name \ cluster_name". See attached mockup. I think the places that will be touched by this are the infra mapping wizard (lots of instances) and the infra mappings list expansion view on the Overview page.

![4aa-createmapping-clusters w datacentre - option 2](https://user-images.githubusercontent.com/6640236/38519115-be66dc80-3c0c-11e8-9f97-ae5f5619eb13.png)


☝️from @vconzola, adding to this as a record (will be a deviation from original mock)

**Comment by AllenBW** [comment]
@priley86 the cluster source and target count, where is that coming from, not seeing it on the first call, nor the attributes to grab such info (full options dump below)
<img width="739" alt="screen shot 2018-04-10 at 1 01 05 pm" src="https://user-images.githubusercontent.com/6640236/38571705-43478e38-3cbf-11e8-87f3-13ba5148f252.png">

I *believe* it can be found in [the call you mentioned above](https://github.com/priley86/miq_v2v_ui_plugin/issues/152#issuecomment-379787249)... but that kinda doesn't jive well with fetching the data when the dropdown is clicked... would need it when overview is loaded 🤔

**Comment by priley86** [comment]
can you get the count from infra mappings query? 
```
export const transformationMappings = [
  {
    id: '10000000000001',
    name: 'Infrastructre Mapping 1',
    description: 'infrastructre mapping 1 description',
    comments: null,
    state: 'draft',
    options: {},
    tenant_id: null,
    created_at: '2018-02-07T16:02:39Z',
    updated_at: '2018-02-07T16:02:39Z',
    transformation_mapping_items: [
      {
        id: '10000000000001',
        source_id: '10000000000002',
        source_type: 'EmsCluster',
        destination_id: '10000000000005',
        destination_type: 'EmsCluster',
        transformation_mapping_id: '10000000000001',
        options: {},
        created_at: '2018-02-07T16:01:44Z',
        updated_at: '2018-02-07T16:02:39Z'
      },
```

**Comment by AllenBW** [comment]
Is the frequency of source and destination  ids the count? if so then yesssssss

**Comment by AllenBW** [comment]
@bzwei wondering how the data from `/api/transformation_mappings` maps to the data from `api/clusters/` ? cuz that dropdown onclick reveals transformation mapping data,  shows all source and target clusters, unclear if info in `transformation_mapping_items` relationship provides a link between the two…

attaching screen shots of a sample of the two objects for reference
<img width="443" alt="screen shot 2018-04-11 at 10 54 09 am" src="https://user-images.githubusercontent.com/6640236/38624681-bc2bc294-3d76-11e8-8142-5ba70c24c942.png">
<img width="501" alt="screen shot 2018-04-11 at 10 53 57 am" src="https://user-images.githubusercontent.com/6640236/38624682-bc3973d0-3d76-11e8-978a-2dd6dc62bc9f.png">

**Comment by AllenBW** [comment]
@vconzola @priley86 after talking with @bzwei I have a few questions for ya'll

Source clusters are always unique, though sometimes map to the same destination clusters

1. the destination cluster count in the row, is this the total number of destination clusters or a count of the unique ones? (example: if unique - 2 source clusters mapping to a single destination cluster, would show the numbers 2/1 respectively; if not unique - 2 source clusters mapping to a single destination cluster, would show the numbers 2/2 respectively; i suspect we want to count unique ones...)
2. for the expanded row, do the source and target columns correspond to each other? or are we just listing all all unique source and destination clusters

**Comment by vconzola** [comment]
@AllenBW  Great questions. I hope I can provide equally great answers.  :-) 

1. Correct for destination/target clusters the number shown should be the number of unique ones.

2. When expanding the row, the idea is that the user should be able to relate source cluster(s) to their associated target cluster. In the mockup below, I show this with spacing because I thought that would be simpler than trying to draw lines between them or other types of visualization. But if you have other ideas for how to show the relationship between source-target in the expanded list I'm open to ideas.
![overview - w mappings mappings copy 3 1 mapping w_ list expansion](https://user-images.githubusercontent.com/22823035/38638165-63a264f0-3d9b-11e8-9650-0086e6c6eca6.png)

**Comment by priley86** [comment]
Resolved in #212
